src/components/resumption/include/resumption/last_state.h

```diff
@@ -43,36 +43,19 @@ namespace resumption {
 class LastState {
  public:
   /**
-   * @brief Constructor
-   */
-  LastState(const std::string& app_storage_folder,
-            const std::string& app_info_storage);
-
-  /**
    * @brief Destructor
    */
-  ~LastState();
+  virtual ~LastState() {}
 
   /**
    * @brief Saving dictionary to filesystem
    */
-  void SaveToFileSystem();
+  virtual void SaveToFileSystem() = 0;
 
   /**
-   * @brief public dictionary
-   */
-  Json::Value dictionary;
-
- private:
-  DISALLOW_COPY_AND_ASSIGN(LastState);
-
-  /**
-   * @brief Load dictionary from filesystem
-   */
-  void LoadFromFileSystem();
-
-  std::string app_storage_folder_;
-  std::string app_info_storage_;
+    * @brief Get reference to dictionary
+    */
+  virtual Json::Value& GetDictionary() = 0;
```

Comment:
@LuxoftAKutsan Do you mean this macro: "DISALLOW_COPY_AND_ASSIGN(LastStateImpl);"?